### DISCUSSION TITLE
Error in tap reporter

### DISCUSSION THREAD

**Discussion by evmiguel** [avajs/ava#3322] [Q&A]
When ava times out, I get this error in the tap reporter when the error type is undefined :
```
originalError.name = name; // Restore the original name.
                           ^

TypeError: Cannot set properties of undefined (setting 'name')
```

I believe the issue is coming from this function: https://github.com/avajs/ava/blob/01ec2804ab9db0ab3ef11e3b5b0c5697d68f8bc5/lib/reporters/tap.js#L11

Is there a way to work around this?

**Comment by novemberborn** [comment]
I’d change that to `originalError?.name = name`. PR welcome.

**Reply by evmiguel** [reply]
Thank you @novemberborn. I made a small PR https://github.com/avajs/ava/pull/3325.
